packages/wallet/backend/src/paymentPointer/service.ts

```diff
@@ -132,4 +134,100 @@ export class PaymentPointerService implements IPaymentPointerService {
       active: false
     })
   }
+
+  async getPaymentPointerAfterValidation(
+    userId: string,
+    accountId: string,
+    paymentPointerId: string
+  ) {
+    const paymentPointer = await PaymentPointer.query()
+      .findById(paymentPointerId)
+      .where('accountId', accountId)
+      .where('active', true)
+
+    if (!paymentPointer) {
+      throw new NotFound()
+    }
+
+    // Check if the user owns the payment pointer.
+    // This function throws a NotFoundException.
+    await this.deps.accountService.findAccountById(
+      paymentPointer.accountId,
+      userId
+    )
+
+    return paymentPointer
+  }
+
+  async generateKeyPair(
+    userId: string,
+    accountId: string,
+    paymentPointerId: string
+  ): Promise<{ privateKey: string; publicKey: string }> {
+    await this.getPaymentPointerAfterValidation(
+      userId,
+      accountId,
+      paymentPointerId
+    )
+
+    // Generate the key pair
+    const { privateKey, publicKey, jwk } = generateJWK()
+    await PaymentPointer.query()
+      .findById(paymentPointerId)
+      .patch({
+        jwk: JSON.stringify(jwk),
+        privateKey,
+        publicKey
+      })
+
+    return { privateKey, publicKey }
+  }
+
+  async registerKey(
+    userId: string,
+    accountId: string,
+    paymentPointerId: string
+  ): Promise<void> {
+    const paymentPointer = await this.getPaymentPointerAfterValidation(
+      userId,
+      accountId,
+      paymentPointerId
+    )
+
+    await this.deps.accountService.findAccountById(accountId, userId)
+
+    if (!paymentPointer) {
+      throw new NotFound()
+    }
+
+    const jwkStr = paymentPointer.jwk
+    if (!jwkStr) {
+      throw new Error(`jwk doesnt' exist.`)
+    }
+
+    await this.deps.rafikiClient.createRafikiPaymentPointerKey(
+      this.generateJwk(paymentPointer.accountId, jwkStr),
```

Comment:
Since you could have multiple keys, I think we'd need to have something other or something in addition to just the accountId